### ISSUE TITLE
Vulnerability in node-sass > sass-graph > yargs > cliui > strip-ansi > ansi-regex

### ISSUE THREAD

**Issue by alexarsh** [sass/node-sass#3190]
- NPM version (`npm -v`): 6.14.15
- Node version (`node -v`): v14.18.0
- Node Process (`node -p process.versions`): 

{
  node: '14.18.0',
  v8: '8.4.371.23-node.84',
  uv: '1.42.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.17.2',
  modules: '83',
  nghttp2: '1.42.0',
  napi: '8',
  llhttp: '2.1.3',
  openssl: '1.1.1l',
  cldr: '39.0',
  icu: '69.1',
  tz: '2021a',
  unicode: '13.0'
}

- Node Platform (`node -p process.platform`): darwin
- Node architecture (`node -p process.arch`): x64
- node-sass version (`node -p "require('node-sass').info"`):

node-sass	6.0.1	(Wrapper)	[JavaScript]
libsass  	3.5.5	(Sass Compiler)	[C/C++]

- npm node-sass versions (`npm ls node-sass`): node-sass@6.0.1

There is the following dependencies tree:

─┬ node-sass@6.0.1
│ └─┬ sass-graph@2.2.5
│   └─┬ yargs@13.3.2
│     ├─┬ cliui@5.0.0
│     │ ├─┬ strip-ansi@5.2.0
│     │ │ └── ansi-regex@4.1.0

When ansi-regex@4.1.0 have the following vulnerability issues: 
https://snyk.io/vuln/npm:ansi-regex@4.1.0

Is there a chance that sass-graph@2.2.5 dependency can be updated in order to fix the issue?

**Comment by matt-cote** [comment]
https://github.com/sass/node-sass/pull/3044

**Comment by katannshaw** [comment]
@alexarsh I had the same issue today. After a lot of trial and error I finally noticed this error message:

```
npm WARN EBADENGINE Unsupported engine { package: 'node-sass@6.0.1',
npm WARN EBADENGINE   required: { node: '>=12' },
npm WARN EBADENGINE   current: { node: 'v10.22.1', npm: '7.21.1' } }
```

So I ran `nvm install v14.18.1` to update my version of node to 14 and it fixed the issue for me. Good luck to you.

**Comment by driskell** [comment]
For many projects dart-sass is substantially slower so is not a viable solution in a lot of cases. In some of our foundation-sites projects the incremental compilation moves from 5 seconds to approximately 5 minutes or longer. There is a new dart VM hosted compilation module in the works but its still very much experimental and no webpack integration yet.

Possibly the best approach is to absorb sass-graph module into node-sass in its current state and then update the dependencies in node-sass accordingly? Would allow node-sass to remain in maintenance mode successfully for a while until the other options mentioned above are available.

**Comment by fmaddenflx** [comment]
Any updates on this?  Has any work been done to absorb sass-graph into node-sass as suggested in the previous post?

**Comment by kiskoza** [comment]
I opened #3202 to absorb sass-graph as suggested, but the maintainers haven't responded anything yet. not sure what else I can do...

**Comment by EchoZhaoH** [comment]
`sass-graph` does not seem to be maintained, is there any update to `node-sass`?

**Comment by xzyfer** [comment]
sass-graph@4.0.0 has been released with a patch for this.

**Comment by xzyfer** [comment]
Fixed in 7.0.1.
